packages/typescript/src/tslib.ts

```diff
@@ -1,13 +1,27 @@
-import resolveId, { AsyncOpts } from 'resolve';
+import resolve, { SyncOpts } from 'resolve';
 
-const resolveIdAsync = (file: string, opts: AsyncOpts) =>
-  new Promise<string>((fulfil, reject) =>
-    resolveId(file, opts, (err, contents) => (err || typeof contents === 'undefined' ? reject(err) : fulfil(contents)))
-  );
+// const resolveIdAsync = (file: string, opts: AsyncOpts) =>
+//   new Promise<string>((fulfil, reject) =>
+//     resolveId(file, opts, (err, contents) =>
+//       err || typeof contents === 'undefined' ? reject(err) : fulfil(contents)
+//     )
+//   );
+
+const resolveId = (file: string, opts: SyncOpts) => resolve.sync(file, opts);
 
 /**
  * Returns code asynchronously for the tslib helper library.
  */
-export default function getTsLibPath() {
-  return resolveIdAsync('tslib/tslib.es6.js', { basedir: __dirname });
-}
+export const getTsLibPath = () => {
+  // Note: This isn't preferable, but we've no other way to test this bit. Removing the tslib devDep
+  //       during the test run doesn't work due to the nature of the pnpm flat node_modules, and
+  //       other workspace dependencies that depenend upon tslib.
+  try {
+    // eslint-disable-next-line no-underscore-dangle
+    return resolveId(process.env.__TSLIB_TEST_PATH__ || 'tslib/tslib.es6.js', {
```

Comment:
I wonder if we should pass a flag into `getTsLibPath` instead?

Reply:
Considered it, but because the call to `getTsLibPath` is buried in the call chain, it was more trouble than it was worth. For as ugly as it is, it's effective and doesn't modify the public API.